include/solvers/navier_stokes_base.h

```diff
@@ -338,23 +349,130 @@ class NavierStokesBase : public PhysicsSolver<VectorType>
    * eventually it could be extended to more than one. By default, the fluid
-   * domain is assumed to have a material_id=0 and the rest of the domains have
-   * a material_id>0.
+   * domain is assumed to have a @p material_id=0 and the rest of the domains
+   * have a @p material_id>0.
    *
-   * @param non_zero_constraints If this parameter is true, it indicates that it is the non_zero constraints
-   * which are being constrainted for the solid domain. If this is set to
-   * false, the zero constraints are constrained in the solid domain
+   * @param[in] non_zero_constraints If this parameter is true, it indicates
+   * that non-zero constraints are being constrained for the solid domain. If
+   * this is set to false, homogeneous constraints are constrained in the solid
+   * domain.
    */
   void
   establish_solid_domain(const bool non_zero_constraints);
 
+  /**
+   * @brief Constrain velocity DOFs of a solid cell.
+   *
+   * @param[in] non_zero_constraints If this parameter is true, it indicates
+   * that non-zero constraints are applied in the solid domain. If
+   * this is set to false, homogeneous constraints are applied in the solid
+   * domain.
+   *
+   * @param[in] local_dof_indices Vector of a cell's local DOF indices.
+   *
+   * @param[out] zero_constraints Homogeneous constraints holding object.
+   */
+  void
+  constrain_solid_cell_velocity_dofs(
+    const bool                                 &non_zero_constraints,
+    const std::vector<types::global_dof_index> &local_dof_indices,
+    AffineConstraints<double>                  &zero_constraints);
+
+  /**
+   * @brief Flag DOFs in solid cells.
+   *
+   * @param[out] dofs_are_in_solid Container of global DOF indices located in
+   * solid cells.
+   *
+   * @param[in] local_dof_indices Vector of a cell's local DOF indices.
+   */
+  void
+  flag_dofs_in_solid(
+    std::unordered_set<types::global_dof_index> &dofs_are_in_solid,
+    const std::vector<types::global_dof_index>  &local_dof_indices);
+
+  /**
+   * @brief Check if the cell is a solid.
+   *
+   * @param[in] dofs_are_in_solid Container of global DOF indices located in
+   * solid cells.
+   *
+   * @param[in] local_dof_indices Vector of a cell's local DOF indices.
+   *
+   * @return Boolean indicating if the cell is in a solid (true) or not (false).
+   */
+  bool
+  check_cell_is_in_solid(
+    const std::unordered_set<types::global_dof_index> &dofs_are_in_solid,
+    const std::vector<types::global_dof_index>        &local_dof_indices);
+  /**
+   * @brief Flag DOFs connected to fluid cells.
+   *
+   * @param[out] dofs_are_connected_to_fluid Container of global DOF indices
+   * connected to at least one fluid cell.
+   *
+   * @param[in] local_dof_indices Vector of a cell's local DOF indices.
+   */
+  void
+  flag_dofs_connected_to_fluid(
+    std::unordered_set<types::global_dof_index> &dofs_are_connected_to_fluid,
+    const std::vector<types::global_dof_index>  &local_dof_indices);
+
```

Comment:
Generally we use a [in] [out] logic in the code, I would move the [out] argument to the last and put the const one first